### PR TITLE
[Backend] Create Missing Brand APIs Added

### LINKED ISSUE #82: [Backend] Create Missing Brand APIs
## Description

- [ ] An API to update the Brand Profile. Make sure the required authentication and authorization
- [ ] And API to delete the Brand Profile. Make sure the required authentication and authorization
- [ ] An open API to fetch the list of all brands(make sure not to include their passwords while fetching)
- [ ] A cron Job API to delete the otps saved under `otp` collection which are older than 1 hour

### PR DESCRIPTION
**Pull request by Hemu21**
## Description:

- [x] An API to update the Brand Profile. Make sure the required authentication and authorization
- [x] And API to delete the Brand Profile. Make sure the required authentication and authorization
- [x] An open API to fetch the list of all brands(make sure not to include their passwords while fetching)
- [x] A cron Job API to delete the otps saved under otp collection which are older than 1 hour

All apis are added.

## Issue

[Issue Link](https://github.com/Kajol-Kumari/InfluMart/issues/82) resolve #82

### CODE REVIEW

**Comment by Hemu21** [comment]
@Kajol-Kumari  can you please check.

File: backend/controllers/brandController.js [hunk 1]
```diff
@@ -82,13 +83,107 @@ exports.getProfile = async (req, res) => {
...
+    name: name || undefined,
+    email: email || undefined,
+    category: category || undefined,
+    location: location || undefined,
+    website: website || undefined,
+    description: description || undefined,
+  };
+
+  // Handle password separately
+  if (password) {
```
Comment (Kajol-Kumari) [inline backend/controllers/brandController.js:100]: I wonder if we should allow password update as well in update API or should we create a separate password reset API 'coz there is some extra checks needed for this no like the current password shouldn't match the new password etc

Comment (Hemu21) [reply backend/controllers/brandController.js:100]: Should I remove that password update from here?

Comment (Kajol-Kumari) [reply backend/controllers/brandController.js:100]: yes, please remove it. There should be separate APIs for reset and update password

**Comment by Hemu21** [comment]
@Kajol-Kumari  updated that not same password check.

**Review by Kajol-Kumari — APPROVED** [review]

**Pull request merged**